src/cli.rs

```diff
@@ -0,0 +1,593 @@
+{
+    pub matches:       &'a ArgMatches<'a>,
+    pub muta:          Arc<Muta<Mapping>>,
+    pub db:            Arc<DB>,
+    pub storage:       Arc<S>,
+    pub txs_wal:       Arc<SignedTxsWAL>,
+    pub consensus_wal: Arc<ConsensusWal>,
+
+    pub pin_ef: PhantomData<EF>,
+}
+
+impl<'a, Mapping, DB, S, EF> Cli<'a, Mapping, DB, S, EF>
+where
+    Mapping: 'static + ServiceMapping,
+    DB: 'static + cita_trie::DB,
+    S: 'static + Storage,
+    EF: ExecutorFactory<DB, S, Mapping>,
+{
+    pub fn new(
+        matches: &'a ArgMatches<'a>,
+        muta: Arc<Muta<Mapping>>,
+        db: Arc<DB>,
+        storage: Arc<S>,
+        txs_wal: Arc<SignedTxsWAL>,
+        consensus_wal: Arc<ConsensusWal>,
+    ) -> Self {
+        Self {
+            matches,
+            muta,
+            storage,
+            db,
+            txs_wal,
+            consensus_wal,
+            pin_ef: PhantomData,
+        }
+    }
+
+    pub fn latest_block(&self) -> ProtocolResult<()> {
+        let mut rt = tokio::runtime::Runtime::new().expect("new tokio runtime");
+
+        match self.matches.subcommand() {
+            ("set", Some(cmd)) => {
+                let height = cmd
+                    .value_of("BLOCK_HEIGHT")
+                    .expect("missing [BLOCK_HEIGHT]");
+
+                match u64::from_str_radix(height, 10) {
+                    Ok(height) => rt.block_on(async move { self.latest_block_set(height).await }),
+                    Err(_e) => Err(CliError::Parse.into()),
+                }
+            }
+
+            ("get", Some(_cmd)) => {
+                let res = rt.block_on(async move { self.latest_block_get().await })?;
+                println!("{:?}", res);
+                Ok(())
+            }
+
+            _ => Err(CliError::Grammar.into()),
+        }
+    }
+
+    pub async fn latest_block_set(&self, height: u64) -> ProtocolResult<()> {
+        let last = self.storage.get_latest_block(Context::new()).await?;
+
+        let block = self.block_get(height).await?;
+        let block = match block {
+            Some(blk) => blk,
+            None => return Err(MainError::Other(height.to_string()).into()),
+        };
+        println!(
+            "latest_block_set, block :{}",
+            serde_json::to_string(&block).unwrap()
+        );
+        self.storage.insert_block(Context::new(), block).await?;
+        println!("latest_block set successfully");
+
+        // now remove 'future' blocks
+        for idx in RangeInclusive::new(height + 1, last.header.height) {
+            self.storage.remove_block(Context::new(), idx).await?
+        }
+        Ok(())
+    }
+
+    pub async fn latest_block_get(&self) -> ProtocolResult<Block> {
+        self.storage.get_latest_block(Context::new()).await
+    }
+
+    pub fn block(&self) -> ProtocolResult<()> {
+        let mut rt = tokio::runtime::Runtime::new().expect("new tokio runtime");
+
+        match self.matches.subcommand() {
+            ("set", Some(cmd)) => {
```

Comment:
Does this mean we can replace a block arbitrarily and not do any data correction? This would disrupt the continuity of the blockchain.

Reply:
Will remove 2 'clear' commands for consensus wal and txs wal and add a 'clear' command directly under 'wal'